### ISSUE TITLE
[docs] Include a docs page targeting performance optimization

### ISSUE THREAD

**Issue by michelengelen** [mui/mui-x#14262]
### Search keywords

-

### Related page

grid

### Kind of issue

Other

### Issue description

The docs have a lot of scattered information regarding best practices to improve performance in the grid and other components. We should probably consolidate this into a specific section and cross-link this in the several call-outs we have scattered already!

The question is should we add several pages to each of the components, or a single one that handles all components in one place?

Thoughts @mui/x ?

**Comment by michelengelen** [comment]
Issue inspired by #14259 👍🏼

**Comment by flaviendelangle** [comment]
I think a lot of the performances tips are component-specific (unlike the slots doc).
So I would be in favor of having at the very least a page dedicated to the data grid (the other set of components like the charts and the tree view could benefit from one as well but not sure if the priority is quite the same).

Do we also need a shared page for more general advises?
Maybe, if we find enough valuable information to put in it. 
I'm worried that without the context of a specific component we will quickly be limited to super vague tips like "memoize what you can memoize" (which do not tell you that for some props memoization is almost a requirement and for other props it's almost pointless).

TLDR: I would start with a data grid performance guide since it's the pain point raised in the issue and the one we have a lot of valuable tips to give.

**Comment by alexfauquette** [comment]
> TLDR: I would start with a data grid performance guide since it's the pain point raised in the issue and the one we have a lot of valuable tips to give.

+1 for keeping that component specific. From charts point of view, perf optimisation might be feature related, so the best place for this kind of info seem to be directly in the feature page
